cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -51,53 +51,62 @@ type shimPod interface {
 	// the `shimExecStateRunning, shimExecStateExited` states. If the exec is
 	// not in this state this pod MUST return `errdefs.ErrFailedPrecondition`.
 	KillTask(ctx context.Context, tid, eid string, signal uint32, all bool) error
+
+	// Specifies if this pod is a template
+	IsTemplate() bool
 }
 
-func createPod(ctx context.Context, events publisher, req *task.CreateTaskRequest, s *specs.Spec) (shimPod, error) {
+// TODO (ambarve): The third return value of this function (the template id)
+// is only needed until we implement the late cloning part after that this return
+// value can be removed.
+func createPod(ctx context.Context, events publisher, req *task.CreateTaskRequest, s *specs.Spec) (shimPod, error, string) {
```

Comment:
The error is always the last return value by convention. Even if you plan to remove this later it's best to follow this convention.